src/chapters/az-cli/walkthrough_az-cli.rst

```diff
@@ -0,0 +1,537 @@
+
+In our case the API hosted by the VM only needs the ability to *read* from its KeyVault. It has no need for writing or deletion capabilities. The minimum permissions we need to grant to the VM to support this use case are:
+
+- ``list``: for accessing the names of secrets
+- ``get``: for accessing the individual secret values
+
+Granting VM Access
+^^^^^^^^^^^^^^^^^^
+
+It's now time to issue our final command:
+
+.. sourcecode:: powershell
+    :caption: Windows/PowerShell
+
+    > az keyvault set-policy -n "$KeyVaultName" --object-id "$VmObjectId" --secret-permissions get list 
+
+.. sourcecode:: bash
+    :caption: Linux/BASH
+
+    $ az keyvault set-policy -n "$keyvault_name" --object-id "$vm_object_id" --secret-permissions get list
+
+If everything went well you should get a confirmation output with a new entry under ``properties.accessPolicies`` for our VM that looks like this:
+
+.. sourcecode:: json
+
+    {
+        "applicationId": null,
+        "objectId": "<vm object ID>",
+        "permissions": {
+          "certificates": null,
+          "keys": null,
+          "secrets": [
+            "get",
+            "list"
+          ],
+          "storage": null
+        },
+        "tenantId": "<azure directory ID>"
+    }
+
+Next Step
+=========
+
+Before moving on let's revisit the web portal and see all the resources we created. Look for your CLI walkthrough resource group. Take a few minutes to see how all of the configurations you performed from the CLI resulted in the same resources as the ones you provisioned before. Remember that the CLI and GUI are just *interfaces* for interacting with the central API that backs them. 
+
+After reviewing your resources it's time to clean up after ourselves by deleting the resource group. This will delete all of the resources contained in it so we don't use up our subscription credits. Notice how we don't have to specify the group because it has been set as a default:
+
+.. sourcecode:: powershell
+
+    # when prompted enter y for yes
+    > az group delete
+
+.. todo:: discuss using service principals for CLI use vs logging in? refer to the addition of our personal account in the access policies list
+
+Congratulations on learning a new way of managing your Azure resources. Now that you have tried both the CLI and GUI, which do you prefer and why? Take a moment to consider how all of these steps could be accomplished in a single command by composing them into a script. When you are ready head over to the :ref:`walkthrough_ws-iis` article to learn about provisioning and configuring a new type of VM -- the Windows Server!
```

Comment:
Instead of asking which they prefer and why ask what are the pros and cons of the GUI and CLI. Even though they may have a preference, that pales in comparison to choosing the correct tool for the job at hand.